include/seqan3/io/sequence/sequence_file_out.hpp

```diff
@@ -459,49 +459,6 @@ class sequence_file_out
         return {};
     }
 
-    /*!\brief           Write a seqan3::record to the file.
-     * \tparam record_t Type of the record, a specialisation of seqan3::record.
-     * \param[in] r     The record to write.
-     *
-     * \details
-     *
-     * ### Complexity
-     *
-     * Constant. TODO linear in the size of the written sequences?
-     *
-     * ### Exceptions
-     *
-     * Basic exception safety.
-     *
-     * ### Example
-     *
-     * ```cpp
-     * sequence_file_out fout{"/tmp/my.fasta"};
-     *
-     * auto it = fout.begin();
-     *
-     * for // ...
-     * {
-     *     record<type_list<dna5_vector, std::string>, fields<field::SEQ, field::ID>> r;
-     *
-     *     // ...
-     *
-     *     fout.push_back(r);
-     * }
-     * ```
-     */
-    template <typename record_t>
-    void push_back(record_t && r)
-        requires tuple_like_concept<record_t> &&
-                 requires { requires detail::is_type_specialisation_of_v<remove_cvref_t<record_t>, record>; }
-    {
-        write_record(detail::get_or_ignore<field::SEQ>(r),
-                     detail::get_or_ignore<field::ID>(r),
-                     detail::get_or_ignore<field::QUAL>(r),
-                     detail::get_or_ignore<field::SEQ_QUAL>(r));
-
-    }
-
```

Comment:
Why did you remove this? The overload below it does something else semantically:
It guesses the association values ←→ field-ID based on the current selection of the file. But for seqan::records we don't need to guess, we can just ask the record.
Maybe the tests still pass after this changed because the record and file configuration always matched. But they don't need to! Instead please create an example that illustrates this behaviour 😄

Reply:
> Why did you remove this?

When this function is present, then all `push_back_tuple` tests compile just for the reason that in your tests you use only tuples of size 2 (seq, id). The call `push_back()` **always** executes the `push_back(record)` function, even if you pass a tuple. Compilation fails, whenever the tuple size is not 2, because record expects exactly 2 template args (it uses the 2 tuple types). 

To force the use of `push_back(tuple)`, which works for tuples of arbitrary size and records as well, I deleted the `push_back(record)` function. It also affects the structure_io code, where I usually have tuples of size 3 or 4, and fastq format, where you query also quality as a 3rd type. 

Can you please give me a hint how to fix that in a better way?

Reply:
> The call push_back() always executes the push_back(record) function, even if you pass a tuple. 

Why? There clearly is a check that prevents this:
```
requires { requires detail::is_type_specialisation_of_v<remove_cvref_t<record_t>, record>; }
```

Also `detail::get_or_ignore` with a field as template parameter obviously isn't defined for a regular tuple so this could never compile.

> Compilation fails, whenever the tuple size is not 2, because record expects exactly 2 template args 

Both functions take one template argument so how is the size of the tuple related to anything?

> To force the use of push_back(tuple), which works for tuples of arbitrary size and records as well, 

As I said above, it **does not work** for records that contain a different field composition than the file. Please see this example from the documentation:
https://github.com/seqan/seqan3/blob/master/include/seqan3/io/sequence/sequence_file_out.hpp#L182
We need a test case that tests this behaviour.

> Can you please give me a hint how to fix that in a better way?

What is broken from your point of view?

Reply:
I guess I understand now; please check the two additional tests.